### DISCUSSION TITLE
[firebase_auth] Option to Disable Silent Push Notification for Phone Number + SMS Sign-In on iOS

### DISCUSSION THREAD

**Discussion by rafaelsetragni** [firebase/flutterfire#10875] [Feature request]
Hello everyone!

I have been using the firebase_auth plugin for signing in users via phone number + SMS. However, I'm encountering issues when using it on iOS devices. On iOS, there are two distinct flows for this process:

1. The primary flow involves receiving a silent push notification containing the verificationID.
2. The secondary flow presents the reCaptcha screen to validate the user, similar to the Android implementation.

I am facing compatibility issues with other notification plugins due to the first flow. To address this, I propose adding an option in the info.plist file (or any other suitable configuration) to disable the silent push notification flow. This would force the plugin to always use the reCaptcha flow for iOS devices.

By implementing this change, it would resolve the compatibility issues with silent push notifications in my application. Additionally, it would ensure that both Android and iOS users experience the same authentication flow.

Thank you for considering this suggestion. I believe it would greatly improve the usability and consistency of the firebase_auth plugin across platforms.

**Comment by benlmyers** [comment]
Found a workaround:

```swift
Auth.auth().settings?.isAppVerificationDisabledForTesting = true
```
